README: SQLite specify encryption limitations (#269)
rancher/steve: README.md

calling `server.New` via the `server.Options.SQLCache` boolean option.
Meaning and behavior are the same unless otherwise specified.

> Note that, if SQLite caching of resources is enabled, some of the data
> is encrypted at rest. In particular:

Reviewer: ```suggestion
Note that, if SQLite caching of resources is enabled, some of the data
can be stored in disk, in either encrypted or plain text form based on:
```

>  - by default, Secrets are encrypted

Reviewer: ```suggestion
 - by default, Secrets are always encrypted
```

 - if the environment variable `CATTLE_ENCRYPT_CACHE_ALL` is set to "true",
all resources are encrypted

>  - regardless of the setting, any filterable/sortable columns are stored

Reviewer: ```suggestion
 - regardless of the setting value , any filterable/sortable columns are stored
```
